sql/analyzer/hoist_select_exists.go

```diff
@@ -35,12 +37,30 @@ func hoistSelectExists(
 	scope *Scope,
 	sel RuleSelector,
 ) (sql.Node, transform.TreeIdentity, error) {
+	a.LogNode(n)
+	nodeAliases, err := getTableAliases(n, scope)
+	if err != nil {
+		return nil, transform.SameTree, err
+	}
+
+	// all renamed aliases will be of the form <alias>_<disambiguationIndex++>
+	disambiguationIndex := 0
+	newAliasCreator := func(alias string) string {
```

Comment:
for the same reason I'd maybe prefer this inlined, making an anonymous function every time we run this rule just to run the for loop in one function doesn't seem optimal

Reply:
Reorganizing the code to avoid anonymous function creation.